### PR TITLE
Fix of modal presentation lifecycle in `pageSheet` mode

### PR DESCRIPTION
**Pull request by jirizoudun**
Added Example target into project to test presentation behaviour. This project contains only implementation of modal presentation with two `coordinator` objects and two `UIViewController` objects. 

All following changes refers to `Starter` class and it's extensions.

As a fix I have added already existing implementation of `UIAdaptivePresentationControllerDelegate` as a delegate to a `presentationController` of a controller which presents a new one. 
I have also removed part of a `func navigationController(_ nc: UINavigationController, didShow vc: UIViewController, animated: Bool)` implementation that caused `coordinator` deallocation on iOS systems prior to iOS 13.
As a last thing I have added `isDismissing` property which is set and unset in `UIAdaptivePresentationControllerDelegate` methods. This property is used to guard call of `didStartCompletion` in `func navigationController(_ nc: UINavigationController, didShow vc: UIViewController, animated: Bool)` because this delegate method is called when user attempts to dismiss a `UIViewController` but doesn't finish and this `UIViewController` stays on top.

### CODE REVIEW

File: ETCoordinator.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -239,7 +390,66 @@
...
 /* Begin XCBuildConfiguration section */
+		52CE7787240015C200CC5FDF /* Debug */ = {
+			isa = XCBuildConfiguration;
+			buildSettings = {
+				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
+				CODE_SIGN_STYLE = Automatic;
+				DEVELOPMENT_TEAM = UVXJ4S52GK;
+				ENABLE_PREVIEWS = YES;
+				INFOPLIST_FILE = ExampleApp/Info.plist;
+				IPHONEOS_DEPLOYMENT_TARGET = 13.2;
```
Comment (urbanp11) [inline ETCoordinator.xcodeproj/project.pbxproj]: framework has target `10.0`... it should be tested on 10, 11, 12 too. But i think we can upgrade target to `11` know.

**Review by urbanp11 — COMMENTED** [review]
`CoordinatorDemo` is not renamed properly (its still in `.pbxproj` file)
line 81 - `7119D2C322952F5500ECA92C /* CoordinatorDemo.swift */ = {isa = PBXFileReference; fileE`

File: ETCoordinator.xcodeproj/project.pbxproj (hunk 1 above)
Comment (urbanp11) [reply ETCoordinator.xcodeproj/project.pbxproj]: It would be nice to have an example app for iOS 11+.

File: Sources/ETCoordinator/Starter.swift [hunk 2]
```diff
@@ -68,24 +72,35 @@ open class Starter: NSObject {
...
     @objc public func navigationController(_ nc: UINavigationController, didShow vc: UIViewController, animated: Bool) {
-        guard isStarted else {
+        guard isStarted && !isDismissing else {
             return
         }
         if let didStartCompletion = didStartCompletion, didStartCompletion.vc === vc {
             didStartCompletion.action()
             self.didStartCompletion = nil
         }
+        // This is called when stoping (poping) pushed coordinator.
```
Comment (jirizoudun) [inline Sources/ETCoordinator/Starter.swift:103]: @urbanp11 
This if caused the memory leak while pushing coordinators. I removed the `didStopWithGestureVC` method call in some of previous commits so now I'm putting it back.

File: ETCoordinator.xcodeproj/project.pbxproj (hunk 1 above)
Comment (jirizoudun) [reply ETCoordinator.xcodeproj/project.pbxproj]: Upgraded to iOS 11

**Pull request merged**